### PR TITLE
check version on request to server - no config

### PR DESCRIPTION
**Pull request by p-nn**

### CODE REVIEW

File: notify.py [hunk 1]
```diff
@@ -34,7 +33,6 @@ def get_service(hass, config, discovery_info=None):
 
     url = config.get(CONF_URL)
     room = config.get(CONF_ROOM)
-    version = int(config.get(CONF_API_VERSION))
 
     try:
         return NextCloudTalkNotificationService(url, username, password, room, version)
```
Comment (noxiii) [inline notify.py]: Hey, 
thank you :)
the version should also be removed in NextCloudTalkNotificationService 
and line 56

File: notify.py [hunk 2]
```diff
@@ -64,9 +62,14 @@ def __init__(self, url, username, password, room, version):
         self._session.headers.update({'Accept': 'application/json'})
         
         """ Get Token/ID for Room """
-        prefix = "/ocs/v2.php/apps/spreed/api/v1"
-        if self.version >= 12:
-          prefix = "/ocs/v2.php/apps/spreed/api/v4"
+        prefix = "/ocs/v2.php/apps/spreed/api/"
+        if self.session.get(self.url+prefix+"v1/room").status_code == 200:
```
Comment (noxiii) [inline notify.py]: The session is protected: self._session

File: notify.py [hunk 3]
```diff
@@ -64,9 +62,14 @@ def __init__(self, url, username, password, room, version):
...
+            self.api_version = 'v1'
+        if self.session.get(self.url+prefix+"v4/room").status_code == 200:
```
Comment (noxiii) [inline notify.py]: the same

**Review by noxiii — COMMENTED** [review]
Hey thank you,
that is what looked today :D and it look good.
I found two small errors.

**Review by p-nn — COMMENTED** [review]
ok

**Review by p-nn — COMMENTED** [review]
ok

File: notify.py (hunk 3 above)
Comment (p-nn) [reply notify.py]: request_rooms = self._session.get(self.url+prefix+"v4/room")
        if request_rooms.status_code != 200:
            request_rooms = self._session.get(self.url+prefix+"v1/room")

File: notify.py (hunk 1 above)
Comment (p-nn) [reply notify.py]: 37: version = int(config.get(CONF_API_VERSION))
40: return NextCloudTalkNotificationService(url, username, password, room, version)
56: def __init__(self, url, username, password, room, version):
60: self.version = version

**Comment by noxiii** [comment]
Thank you :)

**Pull request merged**